### PR TITLE
Refactor backend

### PR DESCRIPTION
**Pull request by geier**

### CODE REVIEW

**Comment by geier** [comment]
It's getting better, thanks to @untitaker suggestion in #320 and not using FLOAT for the ctag in the SQLite DB. I'm periodically getting the same error as @lfam in #309 now. Not really sure where they come from yet.

File: khal/khalendar/khalendar.py [hunk 1]
```diff
@@ -291,36 +113,40 @@ def default_calendar_name(self, default):
...
+            raise ValueError(
+                'Calendar "{0}" is read-only and cannot be used as default'.format(default))
+
+    def _local_ctag(self, calendar):
+        stat = os.stat(self._calendars[calendar]['path'])
+        mtime = getattr(stat, 'st_mtime_ns', None)
+        if mtime is None:
+            mtime = stat.st_mtime
+        import math
+        return str(int(math.floor(mtime * 1e9)))
```
Comment (untitaker) [inline khal/khalendar/khalendar.py]: Two unconditional returns here.

Comment (geier) [inline khal/khalendar/khalendar.py]: True, I was experimenting and not finished here. Also I'm still not sure why the `sleep(1)` is needed.

Comment (untitaker) [inline khal/khalendar/khalendar.py]: Sleep shouldn't be necessary at all.

Comment (geier) [inline khal/khalendar/khalendar.py]: but I need to at least introduce a `sleep(0.01)` to reproducible get the test to pass.

**Comment by geier** [comment]
As this seems to work well enough I'll merge this now (I'm sick of rebasing).

**Pull request merged**